### ISSUE TITLE
Site Administrator can not reset any user's password in @@usergroup-userprefs if there is a member of Administrators group

### ISSUE THREAD

**Issue by fulv** [plone/Products.CMFPlone#1293]
Note: I labeled this as a regression, because it does not happen in Plone 4.3.7.

  1. Create a Plone 5.0 site
  2. Add a user and add them to the *Administrators* group
  3. Add another user and make them a Site Administrator (does not matter whether through direct role or membership in the Site Administrators group)
  4. Log in as the Site Administrator
  5. Go to @@usergroup-userprefs
  6. Check the reset password box for the Site Administrator user, click Save

Result:
See traceback below.

The reason for this problem is in `Products.CMFPlone-5.0-py2.7.egg/Products/CMFPlone/controlpanel/browser/usergroups_usersoverview.py(175)manageUser()`:

```
                roles = user.get('roles', [])
                if not self.is_zope_manager:
                    # don't allow adding or removing the Manager role
                    if ('Manager' in roles) != ('Manager' in current_roles):
                        raise Forbidden
```

The `roles` array for the Manager user does not contain `Manager`, while `current_roles` does.  That's because in the form that user's row has an image under the Manager column, but no checkbox, so the `Manager` role is not added to the POST.

2015-12-21 12:47:36 ERROR Zope.SiteErrorLog 1450730856.150.395936448296 http://localhost:8086/Plone2/@@usergroup-userprefs
Traceback (innermost last):
  Module ZPublisher.Publish, line 138, in publish
  Module ZPublisher.mapply, line 77, in mapply
  Module Products.PDBDebugMode.runcall, line 70, in pdb_runcall
  Module ZPublisher.Publish, line 48, in call_object
  Module Products.CMFPlone.controlpanel.browser.usergroups_usersoverview, line 40, in __call__
  Module Products.CMFPlone.controlpanel.browser.usergroups_usersoverview, line 175, in manageUser
Forbidden
> /Users/fulvio/Downloads/Plone-5.0/buildout-cache/eggs/Products.CMFPlone-5.0-py2.7.egg/Products/CMFPlone/controlpanel/browser/usergroups_usersoverview.py(175)manageUser()
-> raise Forbidden

**Comment by fulv** [comment]
(The workaround is to remove the Manager users from the Managers group, and give them the Manager role directly.)

**Comment by fulv** [comment]
Also, the steps above are contrived for brevity, but of course the real problem is that the Site Administrator can not reset **any other user's password**, either, not just their own.

**Comment by mauritsvanrees** [comment]
Confirmed.

It wasn't clear to me what was going on here. So let me give a bit more information or say it in a different way. That at least makes it clearer to me. :-)

So, what happens is: you want to reset the password of a user, so you check the relevant box and submit the form, and *all* users are submitted, or at least the ones on this page. In my case, the users submitted are these:

```
[{'id': 'admingroup', 'roles': ['Member']}, {'id': 'siteadmin', 'resetpassword': 'siteadmin', 'roles': ['Member']}]
```

The code goes through all users including the admingroup user that you have made no changes to. But Plone see that the roles for this user are changed: Manager is removed. And as Site Administrator this is not allowed, because you do not have the Manager role yourself. So the Forbidden error is raised while checking the admingroup user that we did not actually want to change anything for.

Easiest is probably to have a hidden input in this case. There is code for that in the template, but I think the condition is wrong. Let met make a pull request.

**Comment by fulv** [comment]
Yup, that's exactly it.  Can this issue be closed?
Thanks!

**Comment by idgserpro** [comment]
This happened in the past as well in Plone 4.3.x < 4.3.8, if you changed a Manager's "Login name" in the ZMI, and this user was in the Administrators group in control panel, and you logged with a Site Administrator afterwards trying to edit your users, you got almost the same traceback:

```
2016-04-27 11:39:59 ERROR Zope.SiteErrorLog 1461767999.990.0453734996228 http://localhost:8080/Plone/@@usergroup-userprefs
  Traceback (innermost last):                                                     
    Module ZPublisher.Publish, line 138, in publish                               
    Module ZPublisher.mapply, line 77, in mapply                                  
    Module ZPublisher.Publish, line 48, in call_object                            
    Module plone.app.controlpanel.usergroups, line 194, in __call__               
    Module plone.app.controlpanel.usergroups, line 357, in manageUser             
  Forbidden
```

This was solved in 4.3.8 though.
